src/main/java/com/evgeniyfedorchenko/hogwarts/repositories/FacultyRepository.java

```diff
@@ -1,13 +1,14 @@
 package com.evgeniyfedorchenko.hogwarts.repositories;
 
-import com.evgeniyfedorchenko.hogwarts.models.Color;
-import com.evgeniyfedorchenko.hogwarts.models.Faculty;
+import com.evgeniyfedorchenko.hogwarts.entities.Color;
+import com.evgeniyfedorchenko.hogwarts.entities.Faculty;
 import org.springframework.data.jpa.repository.JpaRepository;
 
 import java.util.List;
 
 public interface FacultyRepository extends JpaRepository<Faculty, Long> {
 
     boolean existsByName(String name);
-    List<Faculty> findByColor(Color color);
+
+    List<Faculty> findFacultyByColorOrNameContainsIgnoreCase(Color color, String name);
```

Comment:
```suggestion
    List<Faculty> findFacultyByColorContainsIgnoreCaseOrNameContainsIgnoreCase(Color color, String name);
```
иначе только по имени будет без учёта регистра